### ISSUE TITLE
Support language variants (en_GB, en_US)

### ISSUE THREAD

**Issue by tbm** [facelessuser/pyspelling#120]
```
  aspell:
    lang: en_GB
```

leads to:

```
  File "/home/tbm/.local/lib/python3.7/site-packages/pyspelling/util/__init__.py", line 89, in get_process_output
    raise RuntimeError("Runtime Error: %s" % (output[0].rstrip().decode(encoding, errors='replace')))
RuntimeError: Runtime Error: Error: The language "en_US" is not known. This is probably because: the file "/usr/lib/aspell/en_US.dat" can not be opened for reading.
```

`aspell -d en_GB` works fine.

It seems `en_XX` files end in `.multi` rather than `.dat`:

```
-rw-r--r-- 1 root root 86 Apr 17  2018 /usr/lib/aspell/en_AU.multi
-rw-r--r-- 1 root root 86 Apr 17  2018 /usr/lib/aspell/en_CA.multi
-rw-r--r-- 1 root root 90 Apr 17  2018 /usr/lib/aspell/en_GB.multi
-rw-r--r-- 1 root root 86 Apr 17  2018 /usr/lib/aspell/en_US.multi
```

**Comment by facelessuser** [comment]
`lang` uses the `--lang` option. 

Maybe you need to pass in the `d` option?

```yml
  aspell:
    lang: en
    d: en_GB
```

**Comment by facelessuser** [comment]
I'm fairly certain the above suggestion will handle your issue. Those options are passed directly to aspell.

If I understand correctly, specifically with aspell, `lang` points to the `.dat` file. There is only one for all the `en` variants. We use `lang` to pass in when compiling the word list as well:

```py
            # Compile wordlist against language
            util.call(
                [
                    self.binary,
                    '--lang', lang,
                    '--encoding=utf-8',
                    'create',
                    'master', output
                ],
                input_text=b'\n'.join(sorted(words)) + b'\n'
            )
```

`lang` is not used with the hunspell wordlist as the options are different, and things are handled differently in that spellchecker when dealing with wordlists.

`d` will specify the different `en` variants.

@gir-bot remove S: triage
@gir-bot add T: support

**Comment by facelessuser** [comment]
This may be a good thing for me to describe in the documentation as I doubt you will be the last to run into this issue.

It's funny as I have been simply using `lang: en` which seems to allow variants from GB, US, etc. Adding the additional option of `d: en_US` flags `behaviour` in my documents as the US prefers `behavior`. I hadn't realized I was using the non-US spelling 🙃. So my suggestion indeed appears to work as intended.

**Comment by tbm** [comment]
Thank you!  I can confirm this works.

The reason I didn't try `d` is because the documentation says that `d` is for hunspell only whereas aspell needs `lang`.

https://facelessuser.github.io/pyspelling/configuration/

> For Aspell, you would use the command line option --lang or -l,

> For Hunspell, you would use the command line option -d,

**Comment by facelessuser** [comment]
Well the docs aren't wrong...they just aren't completely correct either 🙂 . You can pass any of aspell's options through (though we do filter out options that are not helpful in the pyspelling environment).

**Comment by facelessuser** [comment]
Experimenting with this a bit more, it appears, if not using a custom wordlist in Aspell, only `d` is needed, but you could specify `lang` instead, but you will cover all of `en` instead of a region specific `en` like `en_US` or `en_GB`.

```yaml
  aspell:
    d: en_GB
```

But `lang` must be specified for custom wordlists as we must compile the wordlist against the `en.dat` file which is the base of both `en_US`, `en_GB` and other `en` languages. This same logic applies to other languages as well:

```yaml
  aspell:
    d: en_GB
  dictionary:
    # en by default, but manually specified here for clarification
    lang: en
    wordlists:
    - docs/src/dictionary/en-custom.txt
    output: build/dictionary/mkdocs.dic
```

In short, this was simply a misunderstanding on my part when using Aspell (Aspell's documentation isn't always the clearest). This probably would have become more apparent to me early on had I needed to specify more region specific dictionaries, but since I wasn't, I kept operating under incorrect assumptions.

Documentation will be updated in #126 to better explain this. Additionally we've updated our tests and our own personal `.pyspelling.yml` to also reflect this.

**Comment by tbm** [comment]
Looks great!  Thank you.
